main-settings/src/main/scala/sbt/std/TaskLinterDSL.scala

```diff
@@ -66,6 +67,13 @@ abstract class BaseTaskLinterDSL extends LinterDSL {
         } else ()
       }
 
+      def detectAndErrorOnKeyMissingValue(s: Select): Unit = {
```

Comment:
instead of overloading you could just make the other one take a tree

Reply:
This code uses `Select#name`.

Reply:
I guess I can do pattern matching.

Reply:
oh, for some reason I thought all trees had name??